homeassistant/components/nextcloud/config_flow.py

```diff
@@ -0,0 +1,80 @@
+"""Config flow to configure the Nextcloud integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from nextcloudmonitor import NextcloudMonitor, NextcloudMonitorError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+DATA_SCHEMA_USER = vol.Schema(
+    {
+        vol.Required(CONF_URL): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+_LOGGER = logging.getLogger(__name__)
+
+
+class NextcloudConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a Nextcloud config flow."""
+
+    VERSION = 1
+
+    def _try_connect_nc(self, user_input: dict) -> NextcloudMonitor:
+        """Try to connect to nextcloud server."""
+        return NextcloudMonitor(
+            user_input.get(CONF_URL),
+            user_input.get(CONF_USERNAME),
+            user_input.get(CONF_PASSWORD),
```

Comment:
These appear to be compulsory:
```suggestion
            user_input[CONF_URL],
            user_input[CONF_USERNAME],
            user_input[CONF_PASSWORD],
```